src/base/utils/fs.cpp

```diff
@@ -128,18 +132,24 @@ bool Utils::Fs::smartRemoveEmptyFolderTree(const QString &path)
               , [](const QString &l, const QString &r) { return l.count('/') > r.count('/'); });
 
     for (const QString &p : asConst(dirList)) {
-        // remove unwanted files
-        for (const QString &f : deleteFilesList) {
-            forceRemove(p + f);
-        }
-
-        // remove temp files on linux (file ends with '~'), e.g. `filename~`
         const QDir dir(p);
+        if (!dir.isEmpty(QDir::AllDirs | QDir::NoDotAndDotDot))
+            continue;
+
         const QStringList tmpFileList = dir.entryList(QDir::Files);
-        for (const QString &f : tmpFileList) {
-            if (f.endsWith('~'))
+
+        // deleteFilesList contains unwanted files, usually created by the OS
+        // temp files on linux usually end with '~', e.g. `filename~`
+        const bool hasOtherFiles = std::any_of(tmpFileList.cbegin(), tmpFileList.cend(), [&deleteFilesList](const QString &f)
+        {
+            return (!f.endsWith('~') && !deleteFilesList.contains(f, Qt::CaseInsensitive));
+        });
+        if (hasOtherFiles)
+            continue;
+
+        for (const QString &f : tmpFileList)
+            if (f.endsWith('~') || deleteFilesList.contains(f, Qt::CaseInsensitive))
```

Comment:
Why this additional check? hasOtherFiles is false here so there's no "other" files that shouldn't be deleted, isn't it?

Reply:
Sorry. This was lost during the many refactorings. Fixed now.